gflownet/utils/batch.py

```diff
@@ -0,0 +1,288 @@
+        self._process_trajectory_indices()
+        # process other variables, if we are in the train mode and recorded them
+        if len(self.actions) > 0:
+            self.actions = tfloat(
+                self.actions, device=self.device, float_type=self.float
+            )
+            self.done = tbool(self.done, device=self.device)
+            self.masks_invalid_actions_forward = tbool(
+                self.masks_invalid_actions_forward, device=self.device
+            )
+            if self.loss == "flowmatch":
+                self.parents_state_idx = tlong(
+                    sum([[idx] * len(p) for idx, p in enumerate(self.parents)], []),
+                    device=self.device,
+                )
+                self.parents_actions = torch.cat(
+                    [
+                        tfloat(x, device=self.device, float_type=self.float)
+                        for x in self.parents_actions
+                    ]
+                )
+            elif self.loss == "trajectorybalance":
+                self.masks_invalid_actions_backward = tbool(
+                    self.masks_invalid_actions_backward, device=self.device
+                )
+            self._process_parents()
+        self.is_processed = True
+
+    def _process_states(self):
+        """
+        Converts self.statees from list to torch tensor, computes states in the policy format
+        (stored in self.states_policy)
+        """
+        self.states_policy = self.states2policy()
+        self.states = tfloat(self.states, device=self.device, float_type=self.float)
+
+    def states2policy(self, states=None, env_ids=None):
+        """
+        Converts states from a list of states in gflownet fromat to a tensor of states in policy format
+        states: list of gflownet states,
+        env_ids: list of env ids indicating which env corresponds to each state in states list
+
+        Returns: torch tensor of stattes in policy format
+        """
+        if states is None:
+            states = self.states
+            env_ids = self.env_ids
+        elif env_ids is None:
+            # if states are provided, env_ids should be provided too
+            raise Exception(
+                "env_ids must be provided to the batch for converting provided states to the policy format"
+            )
+        states_policy = []
+        for state, env_id in zip(states, env_ids):
+            states_policy.append(self.envs[env_id].state2policy(state))
+        return tfloat(states_policy, device=self.device, float_type=self.float)
+
+    def _process_parents(self):
+        """
+        Prepares self.parents (gflownet format) and self.parents_policy (policy format) as torch tensors.
+        Different behaviour depending on self.loss:
+            - for flowmatch, parents contain all the possible parents for each state,
+            so this tensor is bigger than self.state (all the parents are alinged the zero dimension, )
```

Comment:
aligned
And I think that part in parenthesis is missing a word.

Reply:
sorry, I didn't get what you mean, could you elaborate?

Reply:
Sorry, I meant that : 
- "alinged" should be changed to "aligned" (at least, that what I think it's meant to say)
- that it seems like one or a few words are missing between "all the parents are aligned" and "the zero dimension". 
- finally, the content of the parenthesis ends with a comma. I wasn't sure if it was a typo or an indication that there should be some text following the comma and it was forgotten/mistakenly deleted.

Reply:
Sorry, I meant that : 
- "alinged" should be changed to "aligned" (at least, that what I think it's meant to say)
- that it seems like one or a few words are missing between "all the parents are aligned" and "the zero dimension". 
- finally, the content of the parenthesis ends with a comma. I wasn't sure if it was a typo or an indication that there should be some text following the comma and it was forgotten/mistakenly deleted.